### PR TITLE
Feature/testing result evaluation

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose
- add a blacklist of testsuites that don't have setup durations
- add possibility to filter for more than one test using `--test`
- when using `auto` to locate tests detect whether the tests in questions are `--cluster` only and enable cluster; for a default single server setup one would only get an error in this case
- fix measuring of test setup/teardown times; The values wouldn't be reset after the .js-cache has parsed the jsunity module the first time

The PR currently directs into `bug-fix/speedup-tests` since it also contains these changes. The C++ changes are some merge side effect and not part of this PR.

- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification
This fixes & reworks parts of the test result evaluation framework.
It can be observed by inspecting the measurements of testrus.

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6531/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6552/

**Comment by dothebart** [comment]
tests green.

**Comment by dothebart** [comment]
Tests green.

File: js/client/modules/@arangodb/result-processing.js [hunk 1]
```diff
@@ -547,7 +547,9 @@ function locateLongRunning(options, results) {
             count: Object.keys(testSuite).filter(testCase => skipInternalMember(testSuite, testCase)).length
           });
       } else {
-        print(RED + "This test doesn't have a duration: " + testSuiteName + "\n" + JSON.stringify(testSuite) + RESET);
+        if (!testSuite.hasOwnProperty('skipped') && !testSuite.skipped) {
```
Comment (jsteemann) [inline js/client/modules/@arangodb/result-processing.js]:
```suggestion
        if (!testSuite.hasOwnProperty('skipped') || !testSuite.skipped) {
```

**Pull request merged**